JavaScript SPA Content

andretms opened the pull request:
PR to allow review of the JavaScript SPA content

andretms committed:
Forcing PR

andretms commented:
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md](https://github.com/andretms/quickstarts-test/blob/review/articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C7%5C2%5C58599950-09d3-b0f6-1b9c-85cb52a0ce7a%5CPullRequest%5C201807022344387901-2%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=hq62PUVcAqmEsAZKrRIxkbFMGxwy4S%2BhQCZMyFrKqq0%3D&st=2018-07-02T23%3A40%3A41Z&se=2018-08-02T23%3A45%3A41Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

manoj-rath commented on articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md:
```diff
@@ -0,0 +1,159 @@
+>
+> 1. Go to the [Azure Portal - Application Registration [Prod]](https://portal.azure.com/signin/index/?Microsoft_AAD_RegisteredApps=true#blade/Microsoft_AAD_RegisteredApps/applicationsListBlade/quickStartType/JavascriptSpaQuickstartPage/sourceType/docs)
+> 1. Enter a name for your application and click **Register**
+> 1. Follow the instructions to download and configure your new application, adding **http<span/>://localhost:30662/** as Redirect URL and setting the **Implict Flow** automatically for you.
```
This sentence is confusing.

andretms replied:
@manoj-rath - I changed the sentence - please check

nehaagrawal commented on articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md:
```diff
@@ -0,0 +1,159 @@
+
+### Step 2: Download your web server or project
+
+- [Download the Visual Studio project](https://github.com/Azure-Samples/active-directory-javascript-graphapi-v2/archive/VisualStudio.zip)
```
I think we shouldn't have any VS dependency. We have created dev samples in MSAL that doesn't require any Visual Studio dependency and just requires node.js.

asnow003 replied:
I agree with this...when the first thing I see is "download the VS project" I get turned off right away feeling...oh its complicated ;-).  I think we should have it but we have a nodejs version as well.

andretms replied:
@asnow003 - check my comment below

nehaagrawal commented on articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md:
````diff
@@ -0,0 +1,159 @@
+The quickstart code also shows how you how you initialize the library:
+
+```javascript
+var userAgentApplication = new Msal.UserAgentApplication(msalconfig.clientID, null, loginCallback, {
````
can we explain here what 2nd parameter means? It's the authority url and if user doesn't pass anything then default is https://login.microsoftonline.com/common.

andretms replied:
Good point will do

andretms replied:
Check below now

nehaagrawal commented on articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md:
```diff
@@ -0,0 +1,159 @@
+
+### *msal.js*
+
+MSAL is the library used to sign in users and request tokens used to access an API protected by Microsoft Azure Active Directory. The Quickstart's *index.html* contains reference to the library:
```
can we also mention about npmjs? That way user will know that it's available on both CDN and npmjs even though the samples uses it from CDN .

andretms replied:
@nehaagrawal - will add it

andretms replied:
@nehaagrawal - can you check below?

nehaagrawal commented on articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md:
```diff
@@ -0,0 +1,159 @@
+
+### Requesting tokens
+
+Msal has two methods to used acquire tokens `acquireTokenRedirect` (or `acquireTokenPopup`) and `acquireTokenSilent`:
```
Can we reword it? Even though we say 2 methods there are actually 3 and this can be confusing. We may categorize it as  'Acquire Token Interactive APIs' (acquireTokenRedirect and acquireTokenPopup ) and 'Acquire Token Silent API' (acquireTokenSilent)

nehaagrawal commented on articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md:
```diff
@@ -0,0 +1,159 @@
+### Step 2: Download your web server or project
+
+- [Download the Visual Studio project](https://github.com/Azure-Samples/active-directory-javascript-graphapi-v2/archive/VisualStudio.zip)
+- [Download the core project files - for a local web server, such as Python](https://github.com/Azure-Samples/active-directory-javascript-graphapi-v2/archive/core.zip)
```
why do we need 2 samples?

andretms replied:
@asnow003 , @nehaagrawal - the reason is that our audience is disproportionally skewed towards Visual Studio audiences. Having a Visual Studio project ready-to-go helps people that use Visual Studio. In fact, on our [Guided Setup](https://docs.microsoft.com/azure/active-directory/develop/guidedsetups/active-directory-javascriptspa), clickthrough  rates on the Visual Studio option is more than 100% over the *non-Visual Studio* version. I believe having a Visual Studio option in addition to just the project files provides a better user experience

nehaagrawal replied:
This is a typescript library and users will need Visual studio only if their web service is written in Visual studio. If we are demonstrating the use of MS Graph in our sample, it doesn't require Visual Studio. I am still not convince why we need visual studio.

andretms replied:
VS is just an option to make sure VS users hava an easy path to build and run: for example, if I am a VS user, I can download a zip file, extract it, load in Visual Studio, configure, press <F5> and experiement my app. IF we don't have a VS option, a VS developer would need to do several steps to run the sample app - having an option make it easier for VS users - and like the number suggested, this is the preffered option for most of the users

manoj-rath replied:
After reading all the feedback and response from Andre, I think we should offer two options, in this order
1. using node.js
2. using visual studio

navyasric replied:
I agree with Manoj's feedback here. Even if we offer Visual Studio option, there should be a way for non VS developers to build and run the sample code without having to deal with VS dependency. Node.js does provide a local web server and seems suitable for JS community than Python.

asnow003 commented on articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md:
```diff
@@ -0,0 +1,159 @@
+---
```
General comment...I really like we are doing a simple sample of get token and then get some resources.  It would be nice to see a "composed" sample with all the pieces put together.  Did I miss a link here where you just download a single file and run it?  Just like the old quickstarts from the landing page.
https://identity.microsoft.com/Docs/Web

andretms replied:
Yes, Option 2, Step 2 [here](https://review.docs.microsoft.com/en-us/aad-quickstarts-test/articles/active-directory/develop/quickstarts/active-directory-javascriptspa)?

andretms committed:
Addressing code review comments

andretms commented:
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md](https://github.com/andretms/quickstarts-test/blob/review/articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C7%5C3%5C58599950-09d3-b0f6-1b9c-85cb52a0ce7a%5CPullRequest%5C201807031638004531-2%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=LmerJR9vzGkg%2FtOT%2Fbn25DwYeVxkCGGx33cB8mw224g%3D&st=2018-07-03T16%3A34%3A04Z&se=2018-08-03T16%3A39%3A04Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

andretms committed:
More updates to address code review comments

andretms commented:
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md](https://github.com/andretms/quickstarts-test/blob/review/articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C7%5C3%5C58599950-09d3-b0f6-1b9c-85cb52a0ce7a%5CPullRequest%5C201807031643527115-2%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=EN8R3l3dabH196BMcbVn9MRuLbSfSvFoxFWAwfrvNOM%3D&st=2018-07-03T16%3A39%3A49Z&se=2018-08-03T16%3A44%3A49Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

asnow003 commented on articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md:
```diff
@@ -0,0 +1,172 @@
+---
+title: Azure AD v2 JS SPA Quickstart | Microsoft Docs
```
Another general comment...using "msal".  I talk to customers and I have had people say...what is MSAL/ADAL?  What does that mean?  Looking at the page I don't see a definition to people of what that term is.  Maybe we should stay away from using that in the docs/code?  Like "msalconfig" if I was new to the libraries I might wonder...what is that naming?

andretms replied:
In *More Information* section, there is a description about what MSAL is. About msalconfig - we'd need to modify the code sample - I will start a thread about it.

CelesteDG committed:
Copyedit changes

Made some copyedits including formatting tweaks, small heading updates, corrected a couple of typos, and simplified some sentences/phrases.

CelesteDG commented:
Hello @andretms - I made some tweaks here https://github.com/andretms/quickstarts-test/pull/3 based on the latest version of the file. Please review my suggested changes and approve the PR to your branch if you agree. Thank you.

andretms committed:
Merge pull request #3 from CelesteDG/patch-1

Copyedit changes

andretms commented:
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md](https://github.com/andretms/quickstarts-test/blob/review/articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C7%5C9%5C58599950-09d3-b0f6-1b9c-85cb52a0ce7a%5CPullRequest%5C201807092033538510-2%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=yR8gVLSX4MJwQjOFkx2x5OGa8j8Uvkbr6ruf5ltjBx4%3D&st=2018-07-09T20%3A29%3A49Z&se=2018-08-09T20%3A34%3A49Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

andretms committed:
Additional improvements with JavaScript SPA

andretms commented:
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md](https://github.com/andretms/quickstarts-test/blob/review/articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C7%5C10%5C58599950-09d3-b0f6-1b9c-85cb52a0ce7a%5CPullRequest%5C201807102310033425-2%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=YifxjfWZrIkSckk8UrsmOS694%2BTidNx8FNFvQy%2FYoc8%3D&st=2018-07-10T23%3A05%3A59Z&se=2018-08-10T23%3A10%3A59Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

andretms committed:
Fixing some formatting

andretms commented:
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md](https://github.com/andretms/quickstarts-test/blob/review/articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C7%5C10%5C58599950-09d3-b0f6-1b9c-85cb52a0ce7a%5CPullRequest%5C201807102314318002-2%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=3eOqQz4IRLyxEqE02eul0f4IjkYnHHAhwWOYhFNQUms%3D&st=2018-07-10T23%3A10%3A28Z&se=2018-08-10T23%3A15%3A28Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

navyasric commented on articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md:
```diff
@@ -0,0 +1,178 @@
+> [!div renderon="docs"]
+> ## Register your application and download your quickstart app
+> You have two options to start your quickstart application:
+> * [Express] [Option 1: Register and then download your quickstart application](#option-1-register-your-application-and-download-your-quickstart-app)
```
@andretms I would clarify this as Register and then download the configured sample. Option 2: is register and configure the sample.

navyasric commented on articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md:
```diff
@@ -0,0 +1,178 @@
+> #### Step 1: Register your application
+> 
+> 1. To register an application, go to the [Azure portal - Application Registration [Prod]](https://aka.ms/registeredappsprod) and select **New registration**.
+> 1. Enter a name for your application, add `http<span/>://localhost:30662/` in **Reply URL**, and click **Register**.
```
There is a span tag in the url here by error.

andretms replied:
@navyasric - this is weird by was added on purpose as workaround - because otherwise the markdown converter would automatically translate this to an Hyperlink and we don't want this. I've actually learned that this auto-conversion does not happen if the hyperlink is in between `` - so I am removing this occurrence ;)

navyasric replied:
Oh that's interesting. Didn't know that.

navyasric commented on articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md:
```diff
@@ -0,0 +1,178 @@
+
+> [!div class="sxs-lookup" renderon="portal"]
+> ### Step 1: Configure your application in Azure portal
+> For the code sample for this quickstart to work, you need to add a reply URL as `http<span/>://localhost:30662/` and enable **Implicit Grant**.
```
Span tag in url again.

navyasric commented on articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md:
```diff
@@ -0,0 +1,178 @@
+> > ![Already configured](media/active-directory-windesktop/checkmark.png) Your application is configured with these attributes
+
+> [!NOTE]
+> If you use Visual Studio, the Redirect URL will be set to to *http:<span/>//localhost:30662/* as it is configured in the code sample's project. If you use Python or any other web server, set the redirect URL to `http://<span/>localhost:30662/`, run the following command line:
```
Same as above about the span tag

navyasric commented on articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md:
```diff
@@ -0,0 +1,178 @@
+### Step 3: Configure your JavaScript SPA
+
+> [!div renderon="docs"]
+> Edit `msalconfig.js` and replace `Enter_the_Application_Id_here` with the Application ID your application:
```
@andretms Related to Allen's comment about msalconfig above. I think msalconfig is acceptable file name since MSAL is used in this sample. However, I agree that having some context on MSAL before this step will help. So I suggest in the intro lines for this quickstart we mention that the MSAL.js library is used in the sample.

navyasric commented on articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md:
```diff
@@ -0,0 +1,178 @@
+
+### *msal.js*
+
+MSAL is the library used to sign in users and request tokens used to access an API protected by Microsoft Azure Active Directory (Azure AD). The quickstart's *index.html* contains a reference to the library:
```
You do mention it here but I think mentioning this at beginning will also set context.

navyasric commented on articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md:
````diff
@@ -0,0 +1,178 @@
+
+Alternatively, you can download it through npm:
+
+```batch
````
Curious what is "batch"? Is that supposed to be Bash command line?

navyasric commented on articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md:
```diff
@@ -0,0 +1,178 @@
+
+> |Where  |  |
+> |---------|---------|
+> |`graphAPIScopes`   | (Optional) Contains scopes being requested at login time (that is, `{ "user.read" }` for Microsoft Graph or `{ "api://<Application ID>/access_as_user" }` for custom Web APIs). |
```
@andretms The scopes here should be an array of strings [] and not a hash {}.
Can we also specify that these are requested at login time for consent.

navyasric commented on articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md:
```diff
@@ -0,0 +1,178 @@
+
+> |Where  |  |
+> |---------|---------|
+> |`graphAPIScopes`  | (Optional) Contains scopes being requested at login time (that is, `{ "user.read" }` for Microsoft Graph or `{ "api://<Application ID>/access_as_user" }` for custom Web APIs). |
```
Same comment about scopes being []

navyasric commented on articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md:
```diff
@@ -0,0 +1,178 @@
+
+ There are situations where you need to force users to interact with Azure AD v2.0 endpoint. For example:
+- Users may need to reenter their credentials because their password has expired
+- Your application is requesting access to a resource that the user needs to consent to
```
Can we modify to "Your application is requesting access to additional resource scopes" since v2 is all about scopes mostly.

navyasric commented on articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md:
````diff
@@ -0,0 +1,178 @@
+Alternatively, you can download it through npm:
+
+```batch
+npm install msal
````
Since we are mentioning npm and Node.js. Please add a step saying ensure Node.js is installed with a link to Node.js official installation page.
